framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceImplementation.java

```diff
@@ -1001,6 +1003,33 @@ protected void updateAnnotationSet(
       return;
     }
 
+    // For parameters, If rhsmATM is Nullable or MonotonicNonNull and has the
```

Comment:
This code is nullness/initialization-specific, so `WholeProgramInferenceImplementation` isn't the logical place for it: it should be in the Nullness Checker somewhere.

Reply:
`WholeProgramInferenceImplementation` has checker-specific callbacks, but we can add more if you need another one.